airbyte-lib/airbyte_lib/results.py

```diff
@@ -15,28 +15,39 @@
 
 
 class ReadResult:
-    def __init__(self, processed_records: int, cache: SQLCacheBase) -> None:
+    def __init__(
+        self, processed_records: int, cache: SQLCacheBase, processed_streams: list[str]
+    ) -> None:
         self.processed_records = processed_records
         self._cache = cache
+        self._processed_streams = processed_streams
 
     def __getitem__(self, stream: str) -> CachedDataset:
-        if stream not in self._cache:
+        if stream not in self._processed_streams or stream not in self._cache:
             raise KeyError(stream)
 
         return CachedDataset(self._cache, stream)
 
     def __contains__(self, stream: str) -> bool:
-        return stream in self._cache
+        return stream in self._cache and stream in self._processed_streams
 
     def __iter__(self) -> Iterator[str]:
-        return self._cache.__iter__()
+        return [
+            stream_name
+            for stream_name in self._cache.__iter__()
+            if stream_name in self._processed_streams
+        ].__iter__()
```

Comment:
Since I think this just needs the strings and not the dataset object, can we just iterate over `self._processed_streams` without referencing the cache?

Reply:
theoretically yes, let's go with that for now.